endless-tunnel/README.md

```diff
@@ -0,0 +1,205 @@
+Endless Tunnel
+==============
+Endless Tunnel is a sample game that shows how to:
+- use the Android Studio C++ support
+- implement a game using Android native glue
+- implement joystick support, including robust DPAD navigation for no-touch screens
+
+It is NOT the goal of this sample to show the best way to write the
+game logic, load resources, etc. The game itself was intentionally
+kept rudimentary in order to keep the focus on the Android Studio C++
+integration. For example, this game contains textures and geometry
+hard-coded in code, which works for small demo games like this one,
+but doesn't scale well to real games.
+
+Pre-requisites
+--------------
+- Android Studio 1.3 with [NDK](https://developer.android.com/ndk/) bundle.
+
+Getting Started
+---------------
+1. [Download Android Studio](http://developer.android.com/sdk/index.html)
+1. Launch Android Studio.
+1. Open the sample directory.
+1. Open *File/Project Structure...*
+  1. Click *Download* or *Select NDK location*.
+  1. Click *Tools/Android/Sync Project with Gradle Files*.
+  1. Click *Run/Run 'app'*.
+
+Screenshots
+-----------
+![screenshot](screenshot.png)
+
+Dependencies
+------------
+### GLM LIBRARY
+
+This project uses the OpenGL Mathematics (GLM) library, available at:
+
+    http://glm.g-truc.net/
+
+For more information about this library and its license, please see
+Tunnel/jni/glm/readme.txt and Tunnel/jni/glm/copying.txt.
+
+Acknowledgements
+----------------
+Most of this code was written by [Bruno Oliveira](https://plus.google.com/+BrunoOliveira).
+
+Walkthrough
+-----------
+Source code are under app, it only contains C++ code, which is
```

Comment:
s/are/is needs to be corrected but I think @marcacohen typo'd at the end: "contains" is correct, not containers.